tree/dataframe/src/RDFHelpers.cxx

```diff
@@ -308,7 +342,23 @@ class ProgressBarAction final : public ROOT::Detail::RDF::RActionImpl<ProgressBa
 
    void Exec(unsigned int) {}
 
-   void Finalize() { std::cout << '\n'; }
+   void Finalize()
+   {
+      std::cout << '\n';
+
+      std::mutex fPrintMutex;
+      if (!fPrintMutex.try_lock())
+         return;
+      std::lock_guard<std::mutex> lockGuard(fPrintMutex, std::adopt_lock);
+      std::size_t eventCount;
+      std::chrono::seconds elapsedSeconds;
+      std::tie(eventCount, elapsedSeconds) = fHelper->RecordEvtCountAndTime();
+
+      fHelper->PrintProgressbar(std::cout, eventCount);
+      fHelper->PrintStats_final(std::cout, eventCount, elapsedSeconds);
+
+      std::cout << '\n';
```

Comment:
I would rewrite a bit this function with the following ideas:
* Remove the extra `\n` at the beginning, so we can try to overwrite the same line were the progress bar was previously written
* Avoid running `PrintProgressbar` here, since it's anyway already at the end, we are interested in the final stats mostly
* Use some magic `std::cout << '\r' << std::string(get_tty_size(), ' ') << '\r';` to reset the current output line before writing the final stats. I tested this on my machine, could you also test it on Mac?

```suggestion
      std::mutex fPrintMutex;
      if (!fPrintMutex.try_lock())
         return;
      std::lock_guard<std::mutex> lockGuard(fPrintMutex, std::adopt_lock);
      const auto &[eventCount, elapsedSeconds] = fHelper->RecordEvtCountAndTime();

      // The next line resets the current line output in the terminal.
      // Brings the cursor at the beginning ('\r'), prints whitespace with the
      // same length as the terminal size, then resets the cursor again so we
      // can print on a clean line the final stats.
      std::cout << '\r' << std::string(get_tty_size(), ' ') << '\r';
      fHelper->PrintStats_final(std::cout, eventCount, elapsedSeconds);

      std::cout << '\n';
```